trio/_highlevel_generic.py

```diff
@@ -72,34 +79,35 @@ class StapledStream(HalfCloseableStream, metaclass=Final):
 
     """
 
-    send_stream = attr.ib()
-    receive_stream = attr.ib()
+    send_stream: SendStream = attr.ib()
+    receive_stream: ReceiveStream = attr.ib()
 
-    async def send_all(self, data):
+    async def send_all(self, data: bytes | bytearray | memoryview) -> None:
         """Calls ``self.send_stream.send_all``."""
         return await self.send_stream.send_all(data)
 
-    async def wait_send_all_might_not_block(self):
+    async def wait_send_all_might_not_block(self) -> None:
         """Calls ``self.send_stream.wait_send_all_might_not_block``."""
         return await self.send_stream.wait_send_all_might_not_block()
 
-    async def send_eof(self):
+    async def send_eof(self) -> None:
```

Comment:
Why did you type this as returning `None`? That led to a `# type: ignore`. I think that's totally avoidable!

Reply:
the alternative is `-> Any`, since `self.send_stream.send_eof()` is per definition unknown as it's not defined in `SendStream`. Restricted the `type: ignore` further and added a comment. If `send_eof` is supposed to return something that should be documented somewhere.